ert_shared/ensemble_evaluator/ensemble/builder.py

```diff
@@ -687,19 +691,35 @@ def build(self):
                 self._job_name,
                 self._run_arg,
             )
-        cls = _Step
         if self._type == "function":
-            cls = _FunctionStep
+            return _FunctionStep(
+                stage.get_id(),
+                stage.get_inputs(),
+                stage.get_outputs(),
+                jobs,
+                stage.get_name(),
+                source,
+            )
         elif self._type == "unix":
-            cls = _UnixStep
-        return cls(
-            stage.get_id(),
-            stage.get_inputs(),
-            stage.get_outputs(),
-            jobs,
-            stage.get_name(),
-            source,
-        )
+            return _UnixStep(
+                stage.get_id(),
+                stage.get_inputs(),
+                stage.get_outputs(),
+                jobs,
+                stage.get_name(),
+                source,
+                run_path=self._run_path,
+            )
+        else:
+            return _Step(
+                stage.get_id(),
+                stage.get_inputs(),
+                stage.get_outputs(),
+                jobs,
+                stage.get_name(),
+                source,
+            )
+            # raise AssertionError(f"Unexpected step type: {self._type}")
```

Comment:
The fact that this makes tests fail and that replicating the logic from prior to this PR where a plain `_Step` is returned if it is neither `function` nor `unix` makes me a bit worried... Is this intentional behaviour?

Reply:
Is there a log from this test run, or was it made locally?

Reply:
It was reproduced both locally and in CI. I'll bring it into a separate PR 👍🏻